runbook.tf

```diff
@@ -0,0 +1,52 @@
+############ automation account  + runbook #############
+resource "azurerm_automation_runbook" "vm-start-stop" {
+  name                    = "${var.product}-vm-status-change-${var.env}"
+  location                = var.location
+  resource_group_name     = var.resource_group_name
+  automation_account_name = var.automation_account_name
+  log_verbose             = var.env == "prod" ? "false" : "true"
+  log_progress            = "false"
+  description             = "This is a powershell runbook used to stop and start ${var.product} VMs"
+  runbook_type            = "PowerShell"
+  content                 = file("${path.module}${var.script_name}")
+
+  tags = var.tags
+}
+
+################# automation schedule #################
+resource "azurerm_automation_schedule" "vm-start-stop" {
+  for_each = { for aa_acc_runbook in var.auto_acc_runbooks : aa_acc_runbook.name => aa_acc_runbook }
+
+  name                    = "${var.product}-schedule-${each.value.name}-${var.env}"
+  resource_group_name     = var.resource_group_name
+  automation_account_name = var.automation_account_name
+  frequency               = each.value.frequency
+  interval                = each.value.interval
+  timezone                = var.timezone
+  start_time              = each.value.start_time == null ? "${formatdate("YYYY-MM-DD",timeadd(timestamp(), "24h"))}T19:00:00Z" : "${formatdate("YYYY-MM-DD",timeadd(timestamp(), "24h"))}${each.value.start_time}"
```

Comment:
`run_time` should be mandatory so we should not need to check for null or add a default time

Reply:
Sure. Will udpate.

Reply:
Updated